smslibrary/src/main/java/com/eis/smslibrary/SMSManager.java

```diff
@@ -109,56 +119,78 @@ public void sendMessage(final @NonNull SMSMessage message,
+     * @param context  the context of the application used to setup the listener
+     * @return an {@link ArrayList} of {@link PendingIntent} to be passed to SMSCore.
      */
-    private PendingIntent setupNewSentReceiver(final @NonNull SMSMessage message,
-                                               final @Nullable SMSSentListener listener,
-                                               Context context) {
+    private ArrayList<PendingIntent> setupNewSentReceiver(
+            final @NonNull ArrayList<String> texts, final @Nullable SMSSentListener listener,
+            final @NonNull SMSPeer peer, Context context) {
         if (listener == null || context == null)
-            return null; //Doesn't make any sense to have a BroadcastReceiver if there is no listener or context
+            return null; //Doesn't make any sense to have a BroadcastReceiver if there is no listener
 
-        SMSSentBroadcastReceiver onSentReceiver = new SMSSentBroadcastReceiver(message, listener);
-        String actionName = SENT_MESSAGE_INTENT_ACTION + (messageCounter++);
-        context.registerReceiver(onSentReceiver, new IntentFilter(actionName));
-        return PendingIntent.getBroadcast(context, 0, new Intent(actionName), 0);
+        ArrayList<PendingIntent> intents = new ArrayList<>();
+        IntentFilter intentFilter = new IntentFilter();
+        for (String text : texts) {
+            String actionName = SENT_MESSAGE_INTENT_ACTION + messageCounter++;
+            intents.add(PendingIntent.getBroadcast(context, 0, new Intent(actionName), 0));
+            intentFilter.addAction(actionName);
+        }
+        SMSSentBroadcastReceiver onSentReceiver = new SMSSentBroadcastReceiver(texts, listener, peer);
+        context.registerReceiver(onSentReceiver, intentFilter);
+        return intents;
     }
 
     /**
      * Creates a new {@link SMSDeliveredBroadcastReceiver} and registers it to receive broadcasts
-     * with action {@value DELIVERED_MESSAGE_INTENT_ACTION}
+     * with actions {@value DELIVERED_MESSAGE_INTENT_ACTION}
      *
-     * @param message  that will be sent
-     * @param listener to call on broadcast received
-     * @param context  The context of the application used to setup the listener
-     * @return a {@link PendingIntent} to be passed to SMSCore
+     * @param texts    the parts of the message to be delivered.
+     * @param listener the listener to call on broadcast received.
+     * @param context  the context of the application used to setup the listener
+     * @return an {@link ArrayList} of {@link PendingIntent} to be passed to SMSCore.
      */
-    private PendingIntent setupNewDeliverReceiver(final @NonNull SMSMessage message,
-                                                  final @Nullable SMSDeliveredListener listener,
-                                                  Context context) {
+    private ArrayList<PendingIntent> setupNewDeliverReceiver(
```

Comment:
since this function (and `setupNewSentReceiver` too) works with arrays, you should call them `setupNew___Receivers` (with plural)

Reply:
I left it singular because it sets up a single `BroadcastReceiver`, not multiple receivers.